test/testlib/OpenSSL/Test.pm

```diff
@@ -845,10 +843,22 @@ sub __cwd {
     foreach (@direnv) {
 	if (!file_name_is_absolute($ENV{$_})) {
 	    my $newpath = abs2rel(rel2abs($ENV{$_}), rel2abs($dir));
-	    $ENV{$_} = $newpath;
+	    $tmp_ENV{$_} = $newpath;
 	}
     }
 
+    # Should we just bail out here as well?  I'm unsure.
+    return undef unless chdir($dir);
+
+    if ($opts{cleanup}) {
+	rmtree(".", { safe => 0, keep_root => 1 });
+    }
+
+    %directories = %tmp_directories;
+    foreach (keys %tmp_ENV) {
+        $ENV{$_} = $tmp_ENV{$_};
+    }
+
```

Comment:
The changes in this file look Ok - but I guess I don't understand the problem you described in the commit description. I can't work out what this solves.